### DISCUSSION TITLE
Question regarding Commad line Switch --auto-select-desktop-capture-source="Entire Screen"

### DISCUSSION THREAD

**Discussion by ank-007** [MicrosoftEdge/WebView2Feedback#4577] [Q&A]
chrome when i run it with --auto-select-desktop-capture-source="Entire Screen" does audio share as well but edgewebview2 doesn't why?, In chrome I can turn it off by --screen-capture-audio-default-unchecked, is this flag always enabled in edgewebview2 , i have a webapp which does screensharing running in webview. Any help regarding this is appreciated.  @victorhuangwq  :)

**Comment by victorhuangwq** [comment]
@srwei would you know anything about this?

**Comment by srwei** [comment]
@ank-007  I don't see that Edge does anything differently with respect to that flag. Have you tried if this occurs in the edge browser as well? 
Also, to enable audio on screen capture, you can use the getDisplayMedia() options like navigator.mediaDevices.getDisplayMedia({audio: true}); to enable audio on the preferred display surface.

**Reply by ank-007** [reply]
edge does audio share as well, don't know why edgewebview2 is not able to do it? I'll try gDM and get back , but i was passing --auto-select-desktop-capture-source="Entire Screen" this while launching so it was easier for me,  see this how i am creating and passing args to edgewbview 

std::atomic_flag flag = ATOMIC_FLAG_INIT;
    flag.test_and_set();

    wchar_t currentExePath[MAX_PATH];
    GetModuleFileNameW(NULL, currentExePath, MAX_PATH);
    wchar_t *currentExeName = PathFindFileNameW(currentExePath);

    wchar_t dataPath[MAX_PATH];
    if (!SUCCEEDED(SHGetFolderPathW(NULL, CSIDL_APPDATA, NULL, 0, dataPath))) {
      return false;
    }
    wchar_t userDataFolder[MAX_PATH];
    PathCombineW(userDataFolder, dataPath, currentExeName);

    std::wstring args;
    args.append(L" --ignore-certificate-errors");
    args.append(L" --hide-scrollbars");
	/*ANK 29-01: just leaving this commented, might be useful if we decide to play videos muted
    args.append(L" --mute-audio"); */
    args.append(L" --autoplay-policy=no-user-gesture-required");
    args.append(L" --auto-select-desktop-capture-source=Entire screen");

    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
    options->put_AdditionalBrowserArguments(args.c_str());

    HRESULT res = CreateCoreWebView2EnvironmentWithOptions(
        nullptr, userDataFolder, options.Get(),
        new webview2_com_handler(wnd, cb,
                                 [&](ICoreWebView2Controller *controller) {
                                   m_controller = controller;
                                   m_controller->get_CoreWebView2(&m_webview);
                                   m_webview->AddRef();
                                   flag.clear();
                                 }));
    if (res != S_OK) {
      return false;
    }
    MSG msg = {};
    while (flag.test_and_set() && GetMessage(&msg, NULL, 0, 0)) {
      TranslateMessage(&msg);
      DispatchMessage(&msg);
    }
    init("window.external={invoke:s=>window.chrome.webview.postMessage(s)}");
    return true;

**Reply by ank-007** [reply]
adding to this, I added args.append(L" --use-fake-ui-for-media-stream"); and it is grabbing mic, but i want entire screen audio not mic, getDisplayMedia, I don't know how to override it, since I run external webapps like gmeet, jitsi and teams on webview, I want to select entire screen with audio whenever user tries to do screenshare, with chrome and edge i am able to audio as well.

I suspected it had something to do with audio capturing permission, but i have allowed all the permission kind, 


webview2_com_handler(HWND hwnd, msg_cb_t msgCb,
                         webview2_com_handler_cb_t cb)
        : m_window(hwnd), m_msgCb(msgCb), m_cb(cb) {}
    ULONG STDMETHODCALLTYPE AddRef() { return 1; }
    ULONG STDMETHODCALLTYPE Release() { return 1; }
    HRESULT STDMETHODCALLTYPE QueryInterface(REFIID riid, LPVOID *ppv) {
        (void)riid;
        (void)ppv;
      return S_OK;
    }
    HRESULT STDMETHODCALLTYPE Invoke(HRESULT res,
                                     ICoreWebView2Environment *env) {
      (void)res;
      env->CreateCoreWebView2Controller(m_window, this);
      return S_OK;
    }
    HRESULT STDMETHODCALLTYPE Invoke(HRESULT res,
                                     ICoreWebView2Controller *controller) {
      (void)res;
      controller->AddRef();

      ICoreWebView2 *webview;
      ::EventRegistrationToken token;
      controller->get_CoreWebView2(&webview);
      webview->add_WebMessageReceived(this, &token);
      webview->add_PermissionRequested(this, &token);

      m_cb(controller);
      return S_OK;
    }
    HRESULT STDMETHODCALLTYPE Invoke(
        ICoreWebView2 *sender, ICoreWebView2WebMessageReceivedEventArgs *args) {
      LPWSTR message;
      args->TryGetWebMessageAsString(&message);
      m_msgCb(winrt::to_string(message));
      sender->PostWebMessageAsString(message);

      CoTaskMemFree(message);
      return S_OK;
    }
    HRESULT STDMETHODCALLTYPE
    Invoke(ICoreWebView2 *sender,
           ICoreWebView2PermissionRequestedEventArgs *args) {
      (void)sender;
      COREWEBVIEW2_PERMISSION_KIND kind;
      if (kind == COREWEBVIEW2_PERMISSION_KIND_MICROPHONE ||
            kind == COREWEBVIEW2_PERMISSION_KIND_CAMERA ||
            kind == COREWEBVIEW2_PERMISSION_KIND_UNKNOWN_PERMISSION  ||
            kind == COREWEBVIEW2_PERMISSION_KIND_NOTIFICATIONS ||
              kind == COREWEBVIEW2_PERMISSION_KIND_OTHER_SENSORS ||
              kind == COREWEBVIEW2_PERMISSION_KIND_AUTOPLAY ||
              kind == COREWEBVIEW2_PERMISSION_KIND_WINDOW_MANAGEMENT ||
              kind == COREWEBVIEW2_PERMISSION_KIND_CLIPBOARD_READ) {
          args->put_State(COREWEBVIEW2_PERMISSION_STATE_ALLOW);
        }
      return S_OK;
    }

  private:
    HWND m_window;
    msg_cb_t m_msgCb;
    webview2_com_handler_cb_t m_cb;
  };
}; this is the code, any help is appreciated :)
